src/main.rs

```diff
@@ -49,16 +53,21 @@ impl ProtoBridge {
             | (data as u32 & 0x3fff)
     }
 
+    unsafe fn clock(&self, p_in: *const u8, p_out: *mut u8) {
```

Comment:
Can we just make all the functions take a mutable self instead? They probably should have already... :/